bbq/compiler/compiler.go

```diff
@@ -1730,6 +1732,24 @@ func (c *Compiler[_, _]) compileMethodInvocation(
 	}
 }
 
+func (c *Compiler[_, _]) compileMethodInvocationArguments(
+	invokedExpr *ast.MemberExpression,
+	arguments ast.Arguments,
+	memberInfo sema.MemberAccessInfo,
+	invocationTypes sema.InvocationExpressionTypes,
+) {
+	// Receiver is loaded first. So 'self' is always the zero-th argument.
+	c.compileExpression(invokedExpr.Expression)
+
+	// Unwrap the target, if the member access is via optional chaining.
+	if memberInfo.IsOptional {
+		c.codeGen.Emit(opcode.InstructionUnwrap{})
+	}
```

Comment:
Does this function implement optional chaining (e.g. `v?.m(a, b, ...)`)? Do we need to handle `v == nil` here? In that case the result is `nil`, and neither arguments nor invocation should occur

Reply:
I didn't implement the full optional-chaining feature; The related tests are also still disabled. Added only this fix to get some more unrelated tests working.

I think it would be better to implement the optional-chaining feature separately. We can handle these cases there. WDYT?

Reply:
So does this basically implement enough for the non-nil case of optional chaining to work?
We can/should definitely implement optional chaining in a separate PR if possible 👍 
Could you please leave a TODO here?

Reply:
> So does this basically implement enough for the non-nil case of optional chaining to work?

Yeah, that's sounds about right.

Added a TODO for the full feature: https://github.com/onflow/cadence/pull/3926/commits/067f37ff3813dc754bacfa45c79c2031248e440a